runtime/nvqir/QIRTypes.cpp

```diff
@@ -73,10 +73,6 @@ std::vector<int64_t> getRangeValues(Array *in_array, const Range &in_range) {
   // Convert to absolute index.
   const auto start_idx = convertIndex(in_range.start);
   const auto end_idx = convertIndex(in_range.end);
-  // start == end
-  if (start_idx == end_idx) {
-    return {end_idx};
```

Comment:
Instead of deleting this completely, why not fix it to perform the indirection and return the qubit value? Seems easy enough to do here.
```
      auto qubit = *reinterpret_cast<Qubit **>((*in_array)[end_idx]);
      result.emplace_back(qubit->idx);
      return result;
```

Reply:
Isn't this exactly what happens below, in case `start_idx == end_idx`? Not sure if we are adding any optimization here...

Reply:
Correct.